model/navigation/ProctoringLtiMessageFactory.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA ;
+ */
+declare(strict_types=1);
+
+namespace oat\ltiProctoring\model\navigation;
+
+use InvalidArgumentException;
+use oat\ltiDeliveryProvider\model\navigation\LtiMessageFactoryInterface;
+use oat\oatbox\service\ConfigurableService;
+use oat\taoDelivery\model\execution\DeliveryExecutionInterface;
+use oat\taoLti\models\classes\LtiMessages\LtiMessage;
+use oat\taoProctoring\model\deliveryLog\DeliveryLog;
+use oat\taoProctoring\model\execution\DeliveryExecution as ProctoredDeliveryExecution;
+
+class ProctoringLtiMessageFactory extends ConfigurableService implements LtiMessageFactoryInterface
+{
+    /**
+     * @param DeliveryExecutionInterface $deliveryExecution
+     * @return LtiMessage
+     * @throws \common_exception_NotFound
+     */
+    public function getLtiMessage(DeliveryExecutionInterface $deliveryExecution): LtiMessage
+    {
+        $state = $deliveryExecution->getState()->getLabel();
+        $deliveryLog = $this->getLastDeliveryLogByEvent($deliveryExecution);
+        $ltiLogMessage = $this->prepareLtiLogMessage($deliveryExecution->getState()->getUri(), $deliveryLog);
+
+        return new LtiMessage($state, $ltiLogMessage);
+    }
+
+    private function getEventId(string $executionStateUri): string
+    {
+        switch ($executionStateUri) {
+            case DeliveryExecutionInterface::STATE_FINISHED:
+                $eventId = 'TEST_EXIT_CODE';
```

Comment:
not sure that this question for this PR but, probably, you know why we're using strings instead of constants? (this is not a first place where I see it)

Reply:
Created new PR in taoProctoring with constants for log event IDs - https://github.com/oat-sa/extension-tao-proctoring/pull/812